src/FormTaxonomies/Actions/EnqueueFormBuilderAssets.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Give\FormTaxonomies\Actions;
+
+class EnqueueFormBuilderAssets
+{
+    /**
+     * @var array
+     */
+    protected $settings;
+
+    public function __construct()
+    {
+        $this->settings = give_get_settings();
+    }
+
+    public function __invoke()
+    {
+        if($this->isFormTagsEnabled() || $this->isFormCategoriesEnabled()) {
+            wp_enqueue_script('givewp-builder-taxonomy-settings', GIVE_PLUGIN_URL . 'assets/dist/js/form-taxonomy-settings.js');
+            wp_add_inline_script('givewp-builder-taxonomy-settings','var giveTaxonomySettings =' . json_encode([
```

Comment:
I think usage of `wp_localize_script` would be better suited here so you don't have to manually create a variable and json_encode the data as it does all the for you.

Reply:
My understanding is that `wp_add_inline_script` is the correct way to handle this, as `wp_localize_script` is intended for internationalization.

Also, this avoids some of the "gotchas" such as how localizing a script assumes top level values are strings. Using `wp_add_inline_script` gives more control as to how the data is added to the page.

Reply:
That's fair, apparently WordPress recommends using `wp_add_inline_script` for sending "generic data" to Javascipt.  I've been partial to `wp_localize_script` because the syntax is cleaner but you do technically get more flexiblity with inline.

However, I do think instantiating your variable with `const` rather than `var` would make more sense so it can not be overwritten or declared more than once.

https://developer.wordpress.org/reference/functions/wp_localize_script/#more-information

>Though localization is the primary use, it was often used to pass generic data from PHP to JavaScript, because it was originally the only official way to do that. [wp_add_inline_script()](https://developer.wordpress.org/reference/functions/wp_add_inline_script/) was [introduced in WordPress Version 4.5](https://make.wordpress.org/core/2016/03/08/enhanced-script-loader-in-wordpress-4-5/), and is now the best practice for that use case. `[wp_localize_script()](https://developer.wordpress.org/reference/functions/wp_localize_script/) ` should only be used when you actually want to localize strings.

Reply:
For this case it is important to remember that `wp_localize_script` forces string values and here I'm passing some booleans, which has lead to frustrating bugs in the past. The workaround is to nest non-string values within an array, but at that point might as well just use `wp_add_inline_script`.

Something like `wp_add_inline_data` would be nice, but that would just be a wrapper for `wp_add_inline_script` and `json_encode`.

And yeah, I can make that a `const`. I copy/pasted the structure, which is probably an older example.

Reply:
Resolved, see [aad42d0](https://github.com/impress-org/givewp/pull/7463/commits/aad42d036621164ec5df0d73f78f8a432658519a).